src/pipx/package_specifier.py

```diff
@@ -161,7 +161,27 @@ def parse_specifier_for_install(package_spec: str, pip_args: List[str]) -> Tuple
         )
         pip_args.remove("--editable")
 
-    return (package_or_url, pip_args)
+    for index, option in enumerate(pip_args):
+        if not option.startswith(("-c", "--constraint")):
+            continue
+
+        if option == "-c":
+            next_option_index = index + 1
+            if next_option_index < len(pip_args):
+                next_option = pip_args[next_option_index]
+                pip_args[next_option_index] = str(Path(next_option).expanduser().resolve())
+
+        else:  # option == "--constraint"
+            option_list = option.split("=")
```

Comment:
The `--constraint` option can also be given without the equals sign, apparently.

Reply:
haha, im sorry for not checking this...
 
change this line:

`if option == "-c":`

to this:

`if option == "-c" or option == "--constraint":`